tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -307,50 +315,62 @@ std::vector<TensorView*> mlp(
     local_matmul1 = reshape(local_matmul1, {D, B * S, E}, {D, D, B * S / D, E});
   }
   TensorView* matmul1 = sum(local_matmul1, {0}); // Allreduce or Reduce scatter
-  TensorView* linear1 = add(matmul1, broadcast(b1, {true, true, false}));
+  std::vector<bool> bcast_mask(matmul1->nDims() - 1, true);
+  bcast_mask[matmul1->nDims() - 2] = false;
```

Comment:
Why -2? I could be wrong but I think `matmul1` ends at E not a reduction dimension.

Reply:
hmm...yeah that looks like it should be -1...I'm surprised it worked?

Reply:
oh ... it's because the number of logical dimensions is matmul1->nDims() -1 because there is one reduction axis... so the index of the last one is -2